docs/docs/guides/gitlab.mdx

```diff
@@ -0,0 +1,435 @@
+gitlab's omniauth-connector to ORY Hydra. We'll do that in a docker-based
+lab-environment in order to investigate the details before you do something like
+this in production.
+
+## Preparation
+
+Even though we're mostly using hydra in a docker-container, having the
+command-line-client available is quite useful. So let's install hydra as
+explained in the [installation-guide](../install).
+You'll also need [docker](https://docs.docker.com/get-docker/) and
+[docker-compose](https://docs.docker.com/compose/install/).
+
+The [hydra-5-min-tutorial](../5min-tutorial) might
+be worth checking out upfront. It'll a give a nice quick overview how OAuth2 is
+working within hydra with a minimal example. We assume basic knowledge, here.
+
+If you have not yet the source-code of hydra, which we'll need for the docker-compose
+yaml-files, clone this repository:
+
+```shell
+git clone git@github.com:ory/hydra-gitlab-example.git
+```
+
+## Spin up the instances
+
+Gitlab has some [documentation](https://docs.gitlab.com/omnibus/docker/) about
+how to use their docker-images. It has also an example for docker-compose. So change into the directory with the Hydra 
+source code and create a new file quickstart-gitlab.yml with this content:
+
+```yaml
+version: '3'
+
+services:
+  gitlab:
+    image: gitlab/gitlab-ce:latest
+    restart: always
+    hostname: gitlab.example.com
+    environment:
+      GITLAB_OMNIBUS_CONFIG: |
+        external_url 'http://gitlab.example.com:8000/'
+    ports:
+      - "443:443" # https
+      - "8000:8000" # http
+      - "2222:22" # ssh
+    volumes:
+    - '$GITLAB_HOME/config:/etc/gitlab'
+    - '$GITLAB_HOME/logs:/var/log/gitlab'
+    - '$GITLAB_HOME/data:/var/opt/gitlab'
+```
+
+Now we can spinup the instances.
+
+```shell
+export GITLAB_HOME=$(pwd)/gitlab
+docker-compose -f quickstart.yml \
+    -f quickstart-postgres.yml -f quickstart-gitlab.yml \
+    up --build
+```
+
+Gitlab will take some time until it's up. You'll see the logs in the output. So you need to open a new terminal for the next steps. 
+
+Make sure to have the Domain-Name
+`gitlab.example.com` assigned to `127.0.0.1` before accessing the
+[gitlab-frontend](http://gitlab.example.com:8000). On an unixoid system, this is done by adding
+a line like this to your `/etc/host/file`:
+```
+127.0.0.1 gitlab.example.com
+```
+
+With the initial login, you will have to change the root-password.
+
+## Modify the gitlab configuration for OAuth2
+
+The gitlab-configuration can be modified in the container like such:
+
+```shell
+docker-compose -f quickstart.yml  -f quickstart-postgres.yml -f quickstart-gitlab.yml exec gitlab vim /etc/gitlab/gitlab.rb
+```
+
+Using vim might not be your first choice and if you never used it, it's quite
+unintuitive and confusing but it's worth learning the basics to do simple changes like this. Check the basics
+[here](https://www.youtube.com/watch?v=ggSyF1SVFr4).
```

Comment:
Why not use `nano` instead here? Might help people get this done quicker

Reply:
Yes, good point.